### ISSUE TITLE
[Bug]: time_bucket_gapfill is not gapfilling after upgrade to 2.13.0 

### ISSUE THREAD

**Issue by MA-MacDonald** [timescale/timescaledb#6385]
### What type of bug is this?

Other

### What subsystems and features are affected?

Gapfill

### What happened?

After upgrading from v2.10.3 to v2.13.0  `time_bucket_gapfill` is not filling gaps. 
If there is no data in the time range it returns no rows.

```sql
SELECT
  time_bucket_gapfill('1 day', time) AS date,
  SUM(metric) AS metric
FROM table
WHERE time >= '2021-01-01' AND time < '2022-01-01' 
GROUP BY 1
ORDER BY 1 DESC;
```

Running the above query on v2.10.3 returns the expected output.
![image](https://github.com/timescale/timescaledb/assets/7886459/2d58f8ae-7278-4eb9-8e2e-00257f992f90)

Running the above query on v2.13.0 returns no data.
![image](https://github.com/timescale/timescaledb/assets/7886459/6ddd8bed-4948-4842-937f-ad6ddbd61819)

### TimescaleDB version affected

2.13.0 

### PostgreSQL version used

16

### What operating system did you use?

Windows 10 Enterprise x64

### What installation method did you use?

Not applicable

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
Explained in description
```

**Comment by antekresic** [comment]
Hi there,

I've tried reproducing this issue but was unsuccessful. 

Would you mind giving me a step by step reproduction case so I can try this?

I've tried this on Windows as well but did not get the problem you are getting.

Thanks.

**Comment by MA-MacDonald** [comment]
Hi @antekresic thanks for checking this out.

I did some more testing and when I tried to create a new hypertable I got the following error:
`ERROR: tried calling catalog_get when extension isn't loaded`
Googling led me to these issues https://github.com/timescale/timescaledb/issues/1682, https://github.com/timescale/timescaledb/issues/1844

I did a new backup of the source db on pg15 and restored again on the new db on pg16 and the issue went away.
I'm guessing there was some issue when upgrading from pg15 to pg16.

**Comment by MA-MacDonald** [comment]
Hi @antekresic thanks for checking this out.

I did some more testing and when I tried to create a new hypertable I got the following error:
ERROR: tried calling catalog_get when extension isn't loaded
Googling led me to these issues https://github.com/timescale/timescaledb/issues/1682, https://github.com/timescale/timescaledb/issues/1844

I did a new backup of the source db on pg15 and restored again on the new db on pg16 and the issue went away.
I'm guessing there was some issue when upgrading from pg15 to pg16.
